### ISSUE TITLE
Add dependencies for gradle plugin

### ISSUE THREAD

**Issue by olivierlemasle** [arturbosch/detekt#795]
In a gradle project with multiple modules, I want to add custom Detekt rules. The Detekt plugin is itself a module in my project (e.g. `detekt-extensions`)

It is possible to use Detekt using a gradle plain task:

```groovy
configurations {
    detekt
}

dependencies {
    detekt 'io.gitlab.arturbosch.detekt:detekt-cli:1.0.0.RC6-3'
    detekt project(':detekt-extensions')
}

task detekt(type: JavaExec) {
    main = "io.gitlab.arturbosch.detekt.cli.Main"
    classpath = configurations.detekt
    def input = "$projectDir"
    def config = "$projectDir/detekt.yml"
    def plugins = "detekt-extensions/build/libs/detekt-extensions-${version}.jar"
    def params = [ '-i', input, '-c', config, '-p', plugins]
    args(params)
}
```

However, this is not possible using Detekt gradle plugin (or am I missing something?). To use the extensions with the gradle plugin, I need to compile module `detekt-extensions` manually and configure the path of the JAR file.

It would be great to have the possibility to configure Detekt plugins with Gradle plugin, with only:

```groovy
dependencies {
    detekt project(':detekt-extensions')
}

detekt {
    profile('main') {
        input = projectDir
        config = "$projectDir/detekt.yml"
    }
}
```

**Comment by arturbosch** [comment]
Huh, thats interesting, maybe @rock3r or @tasomaniac know a way to do this?

**Comment by mkobit** [comment]
it is probably because the `detektConfig` configuration is added to a `buildscript.configurations` rather the a `project.configurations` .

https://github.com/arturbosch/detekt/blob/f63b95021593d6793fea02a2db397d2b2a6a6c5f/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt#L22-L23

It might be better if that configuration is created at plugin application time instead of task runtime, and have the extension dependency mapped onto it using the `configurations.withDependencies {}` to lazily populate the `detekt` version

**Comment by olivierlemasle** [comment]
Closed with #845.
